tests/commonTemplates_test.go

```diff
@@ -208,6 +211,42 @@ var _ = Describe("Common templates", func() {
 	})
 
 	Context("resource change", func() {
+		Context("namespace change", func() {
+			BeforeEach(func() {
+				Eventually(func() error {
+					namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: strategy.GetNewTemplatesNamespace()}}
+					return apiClient.Create(ctx, namespaceObj)
+				}, timeout, 10*time.Second).ShouldNot(HaveOccurred())
+				strategy.SkipSspUpdateTestsIfNeeded()
+			})
+
+			AfterEach(func() {
+				Eventually(func() error {
+					namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: strategy.GetNewTemplatesNamespace()}}
+					return apiClient.Delete(ctx, namespaceObj)
+				}, timeout, time.Second).ShouldNot(HaveOccurred())
```

Comment:
This is probably an unstable method to wait for deletion, because when the namespace does not exist, the `Delete()` method will return an error. But it will retry the deletion for 10 minutes, before timing out.

You can use something like:
```go
AfterEach(func() {
	err := apiClient.Delete(ctx, namespace)
	expectSuccessOrNotFound(err)
	waitForDeletion(client.ObjectKey{Name: namespace.Name}, &v1.Namespace{})

	strategy.RevertToOriginalSspCr()
}
```

Reply:
Also, it may be better to call `strategy.RevertToOriginalSspCr()` before deleting the namespace, so the operator will not be confused.